app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/CSVExport.java

```diff
@@ -0,0 +1,42 @@
+package nodomain.freeyourgadget.gadgetbridge.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import nodomain.freeyourgadget.gadgetbridge.devices.SampleProvider;
+import nodomain.freeyourgadget.gadgetbridge.entities.AbstractActivitySample;
+
+/**
+ * Created by Vebryn on 29/05/17.
+ */
+
+public class CSVExport {
+    private static final Logger LOG = LoggerFactory.getLogger(SampleProvider.class);
+
+    public static void exportToCSV(AbstractActivitySample[] activitySamples, File outFile) {
+        String separator = ",";
+
+        LOG.info("Exporting samples into csv file: " + outFile.getName());
+        try (BufferedWriter bw = new BufferedWriter(new FileWriter(outFile))) {
```

Comment:
`FileWriter` always uses the local platform encoding, which, atm, is UTF-8 on Android. Relying on the local platform encoding will in many cases lead to a bug at some point. One has to use the extra verbose
`new OutputStreamWriter(new FileOutputStream(outFile), StandardCharsets.UTF_8))`

I've spent too much of my lifetime on encoding bugs already (I actually have a T-Shirt with the text "Schei? Encoding".

For the record, Excel' CSV import does CP-1252 (similar to ISO-8859-1), LibreOffice can handle all kinds of encodings including UTF-8.